ome/omero-gallery
# remove idr-specific code

In order to use this for other (non-IDR) deployments, we need to remove anything that is specific to IDR:
To test:
 - Go to https://merge-ci.openmicroscopy.org/web/gallery/ (public user OR login first)
 - Should see new gallery UI without any IDR-specific features
 - On IDR ```test70```, with full IDR config and this PR deployed, IDR UI should behave as before with no changes.
 - Going to /gallery-settings/ will show the current gallery config JSON data for ```test70``` or at /gallery/gallery-settings/ for merge-ci (since gallery is not configured at root URL).

With this PR, we need only a single config to see the "new" gallery UI instead of the old one:
e.g.
```
$ bin/omero config set omero.web.gallery.category_queries '{"all":{"label":"All Studies", "index":0, "query":"FIRST50:Name"}}'
```
will list up to 50 Screens and Projects, sorted by Name in the first category.

<img width="1175" alt="Screen Shot 2019-06-26 at 14 34 02" src="https://user-images.githubusercontent.com/900055/60194579-85f8b800-9831-11e9-9d95-4dd45531ce96.png">


To restore IDR functionality we now need the following configs (to be tested at https://github.com/IDR/deployment/pull/189)

```
$ bin/omero config set omero.web.gallery.favicon "https://idr.openmicroscopy.org/about/img/logos/favicon-idr.ico"

$ bin/omero config set omero.web.gallery.top_left_logo '{"src": "http://idr.openmicroscopy.org/about/img/logos/logo-idr.svg"}'

$ bin/omero config set omero.web.gallery.top_right_links '[{"text": "About", "href": "https://idr.openmicroscopy.org/about/index.html", "submenu":[{"text": "Overview", "href": "https://idr.openmicroscopy.org/about/index.html"}, {"text": "Published studies", "href": "https://idr.openmicroscopy.org/about/studies.html"}, {"text": "API Access", "href": "https://idr.openmicroscopy.org/about/api.html"}, {"text": "Data download", "href": "https://idr.openmicroscopy.org/about/download.html"}, {"text": "Image Tools Resource (ITR)", "href": "https://idr.openmicroscopy.org/about/itr.html"}, {"text": "Virtual Analysis Environment (VAE)", "href": "https://idr.openmicroscopy.org/jupyter"}, {"text": "Deployment", "href": "https://idr.openmicroscopy.org/about/deployment.html"}]},{"text": "Submissions", "href": "https://idr.openmicroscopy.org/about/submission.html", "submenu": [{"text": "Overview", "href": "https://idr.openmicroscopy.org/about/submission.html"}, {"text": "Screens", "href": "https://idr.openmicroscopy.org/about/screens.html"},        {"text": "Experiments", "href": "https://idr.openmicroscopy.org/about/experiments.html"}]}]'

$ bin/omero config set  omero.web.gallery.subheading_html "The Image Data Resource (IDR) is a public repository of image datasets from published scientific studies,<br/>where the community can submit, search and access high-quality bio-image data."

$ bin/omero config set omero.web.gallery.footer_html IDR

// Existing settings...
$ bin/omero config set omero.web.gallery.filter_keys '[{"label": "Name (IDR number)", "value": "Name"}, "Imaging Method", "License", "Organism", "Publication Authors", "Publication Title", "Screen Technology Type", "Screen Type", "Study Type"]'

$ bin/ omero config set omero.web.gallery.base_url "https://idr.openmicroscopy.org/"

$ bin/omero config set omero.web.gallery.category_queries '{"latest": {"label": "Most Recent studies", "index": 0, "query": "LAST10:date"}, "timelapse": {"label": "Time-lapse imaging", "index": 1, "query": "Study Type:time OR Study Type:5D OR Study Type:3D-tracking"}, "lightsheet": {"label": "Light sheet fluorescence microscopy", "index": 2, "query": "Study Type:light sheet"}, "proteinlocalization": {"label": "Protein localization studies", "index": 3, "query": "Study Type:protein localization"}, "histology": {"label": "Digital pathology imaging", "index": 4, "query":"Study Type:histology"}, "yeast": {"label": "Yeast studies", "index": 5, "query": "Organism: Saccharomyces cerevisiae OR Organism:Schizosaccharomyces pombe"}, "humancellscreen": {"label": "High-content screening (human)", "index": 6, "query": "Organism:Homo sapiens AND Study Type:high content screen"}}'

$ bin/omero config set omero.web.gallery.study_short_name '[{"key": "Name", "regex": "^(.*?)-.*?(.)$", "template": "$1$2"}]'
```
